cluster-creation-cli/README.md

```diff
@@ -0,0 +1,163 @@
+
+## Basic syntax
+
+The command should be named `create cluster`, hence the entire command, without any flags, would be:
+
+    kubectl gs create cluster
+
+The most important flags would be:
+
+- `--provider`: (required) provider name. For example: capa, capvcd, capz, eks.
+  - TODO: How do we restrict and validate the provider list per installation?
+- `--config`: Path to a config file (optional). Can be used multiple times. See "Configuration layering" below. As an alternative, or in addition, configuration can be passed via standard input.
+- `--app-version`: the cluster-PROVIDER app version to use. If not specified, the latest version is used.
+- `--dry-run`: only validate the configuration and print the resulting manifest.
+- `--output`: file path to write the result to. If not specified, resources are written to the management cluster. The option `STDOUT` can be used to print the result to the console.
+- `--format`: This flag can be used to specify the format of the output.
+  - Default is `manifest` for a Kubernetes manifest that includes an App resources and one or several ConfigMap resources.
+  - `config` for merged configuration YAML.
+  - `command` for a self-contained representation of the command just executed, as a one-liner.
+- `--set`: Override a configuration value. This flag can be used multiple times. The format is `path.to.property=value`.
+  - Note: `--set` can only be used with scalar values (string, number, boolean, integer, etc.). It cannot be used to override an array or object.
+
+TODO:
+
+- How should the resource **namespace** be specified? It could be derived from the organization name included in the cluster config, however that would require logical understanding of the config, which may differ over time and between providers.
+
+### Examples
+
+Ex. 1: Simple cluster creation based on a file
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml
+
+Ex. 2: Simple cluster creation based on standard input
+
+    kubectl gs create cluster --provider capa <<EOF
+    global:
+      metadata:
+        servicePriority: lowest
+        name: test01
+        organization: testorg
+        description: Just a test cluster
+    EOF
+
+While ex. 1 appears simpler and allows for easier re-use of a file-based configuration for a single user, ex. 2 has the benefit of being completely self-contained. It's a pure one-liner without the need for a separate file. This way it's easy to share the command, or copy the command from a website or web UI and paste it into a terminal.
```

Comment:
I can see this getting outdated pretty quickly as the schema changes, how would we maintain it?  Would it need to be included into the release notes?

Reply:
That's an interesting point. In a previous draft document I had an [entire section](https://docs.google.com/document/d/1l_jK3F9ePEe_Jab0UcLO7u_B24bZNbBwwzPA0hSu-LA/edit#heading=h.w3qtd4lxetxk) about this, but decided to skip it here in order to not bloat the RFC. I'll copy it here:

> ## Cluster app schema evolution and incompatibilities
>
> As cluster apps' schemas evolve, we sometimes introduce incompatible changes. For example, we move properties into a different place. With the specification described above, we will encourage users to keep their cluster configuration values YAML files around for re-use, which will eventually lead into users facing incompatibilities.
>
> Most important in this situation is that these incompatibilities get caught through the validation executed by the `kubectl gs create cluster` command. This way, the command will fail with an error. However, this requires that the schema used for validation does not allow arbitrary properties (additionalProperties) on any object.
>
> Outside the scope of this spec, we may think about further improvements to provide a smooth experience when creating clusters using config files form previous app versions. Some ideas:
>
> - Add values migration into the CLI as a separate command, or even
> - Automatically migrate values from previous schemas into the selected app version's schema
> - Have a schema version and require/allow configuration to specify the schema version to validate against

For now, I think that the best approach we had was the `yq` migration command embedded in the changelog ([example](https://github.com/giantswarm/cluster-aws/blob/main/CHANGELOG.md#0500---2023-12-04)).